### DISCUSSION TITLE
How do I change the header title from "Textual Application" to something else?

### DISCUSSION THREAD

**Discussion by jeffwright13** [Textualize/textual#297] [Q&A] [answered]
OK this is driving me nuts. I cannot figure out how to configure/change the Header (or Footer) widgets so they will display a title of my choosing. The code doesn't seem to provide an easy way to do this...there is no "title" attribute in the constructor, nor are there obvious methods by way of which to set the title after instantiation. What am I missing here?

**Comment by willmcgugan** [comment]
Can you set the title attribute in the header?

**Reply by jeffwright13** [reply]
There is no setter that I can see. There is a getter, via the `@property` decorator (which implements a read only attribute). The code accepts setting "title" when I try set via `__setattr__`, but the application itself still shows the header widget with title "Textual Application".

https://github.com/jeffwright13/pytest-fold/blob/issue25/pytest_fold/tuit2.py
(line 52 is the start of the header definition code)

**Reply by willmcgugan** [reply]
Why __setattr__? What happens if you do header.title = “Howdy” ?

**Reply by jeffwright13** [reply]
I used `__setattr__` because I did not think I could set the attribute `title` directly. 

When I set directly via `header.title`, the header widget still displays "Textual Application".
```
    async def on_mount(self) -> None:
        header = Header(tall=False)
        header.title = "bar"
```

**Comment by tusharsadhwani** [comment]
I just peeked at the code, and in `src/textual/app.py`, the `App` class takes a `title` parameter in its constructor.

**Reply by jeffwright13** [reply]
So, set title at the App level and not the Widget level?

**Reply by willmcgugan** [reply]
Both should work.

app.title = “foo”

Or

header.title = “bar”

**Reply by jeffwright13** [reply]
The code I quoted above instantiates the PytestFold app (which is subclassed from App) and specifies a 'title' in the constructor, but the console still shows the default "Textual Application" text. 

I'll try your suggestion and see what happens.

**Reply by jeffwright13** [reply]
No joy. :-( I feel like I"m missing something basic here!

**Comment by jeffwright13** [comment, marked as answer]
I ended up inserting the following line in my app's `on_mount()` method and the header is now showing my title:
```
self.title = "Woof!"
```

**Comment by Green1208** [comment]
Petal to metal
